Src/Base/AMReX_GpuDevice.cpp

```diff
@@ -208,7 +209,11 @@ Device::Initialize ()
     }
     else {
         if (amrex::Verbose() && ParallelDescriptor::IOProcessor()) {
-            amrex::Warning("Multiple GPUs are visible to each MPI rank. This is usually not an issue. But this may lead to incorrect or suboptimal rank-to-GPU mapping.");
+            if ((Machine::name() != "nersc.perlmutter") &&
+                (Machine::name() != "olcf.frontier"))
+            {
+                amrex::Warning("Multiple GPUs are visible to each MPI rank. This is usually not an issue. But this may lead to incorrect or suboptimal rank-to-GPU mapping.");
```

Comment:
I think with the implementation below, we need to be more precise now: for the machines we implement logic for, we should post something like: "Fixing GPU assignment for Frontier according to heuristics..." or so?